scripts/rotate-api-tokens.sh

```diff
@@ -0,0 +1,106 @@
+# to manually approve.
+#
+# If running locally you will need to provide AWS_PROFILE=sops or equivalent.
+# If running locally through Docker, you will probably also need to mount your ~/.aws folder at `/root/.aws` for SOPS to be able to decrypt credentials. But be careful with this due to file permissions.
+# If running on Jenkins, AWS allows SOPS to decrypt credentials using its EC2 instance profile - even within Docker.
+#
+# Syntax: ./scripts/rotate-api-tokens.sh [add-new|remove-old] [preview|staging|production]
+
+set -e
+
+COMMAND=$(echo "${1}" | tr "[:lower:]" "[:upper:]")
+STAGE=$(echo "${2}" | tr "[:lower:]" "[:upper:]")
+STAGE_LOWER=$(echo "${STAGE}" | tr "[:upper:]" "[:lower:]")
+GIT_CREDS_UPDATE_BRANCH_NAME=$(date +"%Y-%m-%dT%H-%M-%S-credentials-update")
+
+if [ -z "${GITHUB_ACCESS_TOKEN}" ]; then
+  echo "Required environment variable GITHUB_ACCESS_TOKEN is undefined."
+  exit 1
+fi
+
+if [[ "${STAGE}" != "PREVIEW" && "${STAGE}" != "STAGING" && "${STAGE}" != "PRODUCTION" ]]; then
+  echo "Syntax: ./scripts/rotate-api-tokens.sh [add-new|remove-old] [preview|staging|production]"
+  exit 2
+fi
+
+function generate_api_token() {
+  echo $(python3 -c "import random, string; print(''.join(random.SystemRandom().choice(string.ascii_letters + string.digits) for _ in range(56)))")
+}
+
+function get_jenkins_env_token_name() {
+  if [[ "$1" == "api" ]]; then
+    app_name="DATA_API"
+  elif [[ "$1" == "search_api" ]]; then
+    app_name="SEARCH_API"
+  else
+    exit 3
+  fi
+
+  echo "DM_${app_name}_TOKEN_${STAGE}"
+}
+
+function create_credentials_update_branch() {
+  rm -rf digitalmarketplace-credentials
+  git clone "https://${GITHUB_ACCESS_TOKEN}@github.com/alphagov/digitalmarketplace-credentials.git"
+  cd digitalmarketplace-credentials
+  git checkout -b "${GIT_CREDS_UPDATE_BRANCH_NAME}"
+}
+
+function commit_and_create_github_pr() {
+  git config user.email "support@digitalmarketplace.service.gov.uk"
+  git config user.name "dm-ssp-jenkins"
+  git commit -a -m "$1" -m "$2"
+  git push origin "${GIT_CREDS_UPDATE_BRANCH_NAME}"
+  post_data="{\"title\": \"$1\", \"body\": \"$2\", \"base\": \"master\", \"head\": \"${GIT_CREDS_UPDATE_BRANCH_NAME}\"}"
+  response_data=$(curl -XPOST -H "Accept: application/vnd.github.v3.full+json" -d "$post_data" "https://${GITHUB_ACCESS_TOKEN}@api.github.com/repos/alphagov/digitalmarketplace-credentials/pulls")
+  echo "Created PR#$(echo "${response_data}" | jq -crM '.number') on alphagov/digitalmarketplace-credentials."
+  echo "https://www.github.com/alphagov/digitalmarketplace-credentials/pull/$(echo "${response_data}" | jq -crM '.number')"
+}
+
+function add_new_tokens() {
+  create_credentials_update_branch
+
+  echo "Adding three new tokens (Application, Jenkins, Developer) to the Data-API and Search-API for ${STAGE}."
+
+  # Insert three new tokens for the api and search-api, and update Jenkins config with the new jenkins token.
```

Comment:
Wonder if it's worth explicitly mentioning that they are being inserted at the top (it is fairly well documented for the removal case that it's not the first three)

Reply:
Ugh missed this comment before putting this in. 🤦‍♂️